tools/jit/gen_jit_dispatch.py

```diff
@@ -229,8 +253,26 @@ def emit_decl(decl):
         'returns': [{'name': 'result', 'type': 'int64_t', 'dynamic_type': 'int64_t'}],
     } for name in ['sizes', 'strides', 'dim']]
     aten_decls = load_aten_declarations(declarations) + tensor_impl_methods
+
     jit_decls = [d for d in aten_decls if is_jit_op(d)]
 
+    # add arguments dtype and device for functions like zeros
+    for decl in jit_decls:
+        arguments = decl['arguments']
+        if len(arguments) == 0 or arguments[0]['simple_type'] != 'Type':
+            continue
+        del arguments[0]
+        arguments.extend([
+            # dtype is specified as an int64_t of at::ScalarType
```

Comment:
I think I don't understand well enough how `torch::ones()` handles devices input. According to native_functions.yaml it has the following signature:

```
- func: ones(Type dtype, IntList size) -> Tensor
  variants: function
```

I don't see how the Type can contain the device information that you're substituting here. According to 5edf6b203749f3088569ccb5a83fbbe3ece97871 this is not actually supported yet. @gchanan/@goldsborough do the ATen factories support being passed devices? Not yet, IIRC?

Maybe this is all fine because we *eventually* want to do this, but it's something that anyone who actually adds this to ATen later will have to be aware of when working with the codegen.

Reply:
Correct `ones` does not set the device. Instead the `CALL_DTYPE` template has an  `AutoGPU` guard that uses the device setting. We do want to put this info into the `ones` op eventually though.

Reply:
yeah this goes away in my PR since I've pushed `AutoGPU` all the way into `TypeDerived::tensor`, the most fundamental allocation source